internal/cmd/schema-registry/command_schema_delete.go

```diff
@@ -67,24 +67,33 @@ func deleteSchema(cmd *cobra.Command, srClient *srsdk.APIClient, ctx context.Con
 		return err
 	}
 
+	permanent, err := cmd.Flags().GetBool("permanent")
+	if err != nil {
+		return err
+	}
+
 	checkVersion := version
 	if version == "all" {
 		// check that at least one version for the input subject exists
 		checkVersion = "latest"
 	}
-	_, httpResp, err := srClient.DefaultApi.GetSchemaByVersion(ctx, subject, checkVersion, nil)
-	if err != nil {
+	if permanent {
+		getOpts := &srsdk.GetSchemaByVersionOpts{Deleted: optional.NewBool(true)}
+		if _, httpResp, err := srClient.DefaultApi.GetSchemaByVersion(ctx, subject, checkVersion, getOpts); err != nil {
+			return errors.CatchSchemaNotFoundError(err, httpResp)
+		} else if _, _, err := srClient.DefaultApi.GetSchemaByVersion(ctx, subject, checkVersion, nil); err == nil {
+			return errors.New("you must first soft delete a schema version before you can hard delete it")
```

Comment:
Wait really? That seems like a silly requirement.

Reply:
Does the backend not return an equivalent error message?

Reply:
Nit: Match the wording with the flag name.
```suggestion
			return errors.New("you must first soft delete a schema version before you can permanently delete it")
```

Reply:
> Wait really? That seems like a silly requirement.

Yeah. If you try to delete one you get the following backend error: `"Subject 'test3' Version 1 was not deleted first before being permanently deleted`.

> Does the backend not return an equivalent error message?

The headache here is that I can't get that error without calling the delete endpoint, and for the purposes of checking existence beforehand the get endpoint returns no error. So I have to infer whether or not it was soft deleted with this unfortunate second call if I know that it wasn't hard deleted after the first call.

Reply:
Can we let the backend team know about this? Maybe they can help fix some error messages?

Reply:
I'll make a ticket asking if they can update the sdk, but otherwise I think this is working as intended. The `deleted` option for GET requests seems to be just "return the version even if it was soft-deleted" and not "return the version only if it was soft-deleted"